query-engine/black-box-tests/tests/metrics/smoke_tests.rs

```diff
@@ -55,22 +55,54 @@ mod smoke_tests {
                 .text()
                 .await
                 .unwrap();
+            
+            // I would have loved to use insta in here and check the snapshot but the order of the metrics is not guaranteed
```

Comment:
We might want to guarantee the order of metrics in the future maybe then?

Reply:
I thought about it, but given that's not a requirement that is set on the exposed format. I would say that sorting wouldn't add any value for consumers of metrics, and is an additional operation to make at runtime, that would (a priori) only benefit testability.

But also given that the values of the metrics will vary from run to run, (in particular histograms recording times) we would need further processing on the response to match it with a snapshot. To the least, we would need setting numeric values to a placeholder before comparing.

We could have another approach to testing, like unmarshaling into a struct and making assertions on it, but in my opinion, doing so won't add better coverage than what we have now.